### PR TITLE
fix(hcm): 2nd passthrough of subfields to make nullable

### PR DESCRIPTION
**Pull request by gunnar-solo**
special care taken with `Duration` and `ResourceRef` object types.
* updated tests

# Feature validation

When combining this `MatchableHttpGateway`
```yaml
apiVersion: gateway.solo.io/v1
kind: MatchableHttpGateway
metadata:
  name: client-ip-reject-gateway
  namespace: gloo-system
spec:
  httpGateway:
    options:
      httpConnectionManagerSettings:
        tracing:
          verbose: false
    virtualServices:
      - name: client-ip-reject
        namespace: gloo-system
  matcher: {}
```

With this `Gateway`
```yaml
apiVersion: gateway.solo.io/v1
kind: Gateway
metadata:
  labels:
    app: gloo
  name: gateway-proxy
  namespace: gloo-system
spec:
  bindAddress: '::'
  bindPort: 8080
  hybridGateway:
    delegatedHttpGateways:
      httpConnectionManagerSettings:
        tracing:
          verbose: true
      ref:
        name: client-ip-reject-gateway
        namespace: gloo-system
  options: {}
  proxyNames:
    - gateway-proxy
  useProxyProto: false
```

you generate this `Proxy`
```yaml
apiVersion: gloo.solo.io/v1
kind: Proxy
metadata:
  creationTimestamp: '2022-09-01T21:18:00Z'
  generation: 12
  labels:
    created_by: gloo-gateway-translator
  name: gateway-proxy
  namespace: gloo-system
  resourceVersion: '3358'
  uid: bcd004ca-9438-46cb-924e-8947a22032e7
spec:
  listeners:
    - bindAddress: '::'
      bindPort: 8080
      hybridListener:
        matchedListeners:
          - httpListener:
              options:
                httpConnectionManagerSettings:
                  tracing:
                    verbose: false
              virtualHosts:
                - domains:
                    - '*'
                  metadataStatic:
                    sources:
                      - observedGeneration: '3'
                        resourceKind: '*v1.VirtualService'
                        resourceRef:
                          name: client-ip-reject
                          namespace: gloo-system
                  name: gloo-system.client-ip-reject
                  routes:
                    - directResponseAction:
                        body: |
                          client ip forbidden
                        status: 403
                      matchers:
                        - prefix: /
                      metadataStatic:
                        sources:
                          - observedGeneration: '3'
                            resourceKind: '*v1.VirtualService'
                            resourceRef:
                              name: client-ip-reject
                              namespace: gloo-system
            matcher: {}
      metadataStatic:
        sources:
          - observedGeneration: '8'
            resourceKind: '*v1.Gateway'
            resourceRef:
              name: gateway-proxy
              namespace: gloo-system
      name: 'listener-::-8080'
      options: {}
      useProxyProto: false
    - bindAddress: '::'
      bindPort: 8443
      httpListener: {}
      metadataStatic:
        sources:
          - observedGeneration: '3'
            resourceKind: '*v1.Gateway'
            resourceRef:
              name: gateway-proxy-ssl
              namespace: gloo-system
      name: 'listener-::-8443'
      useProxyProto: false
status:
  statuses:
    gloo-system:
      reportedBy: gloo
      state: 1
```

Which!  If you dig through all of the fluff:
* `MatchableHttpGateway`: httpConnectionManagerSettings.tracing.verbose=false
* `Gateway`: httpConnectionManagerSettings.tracing.verbose=true
* `Proxy`: httpConnectionManagerSettings.tracing.verbose=false

meaning!  That this correctly solves the 0-value override problem for bools

### CODE REVIEW

**Comment by gunnar-solo** [comment]
/kick [Step #10 - "test": [Fail] Gateway in memory tcp gateway ssl [It] should work with ssl](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=557de15a-09a7-4529-bf02-09898bf9f62a)

File: projects/gateway/pkg/translator/hybrid_translator_test.go [hunk 1]
```diff
@@ -385,171 +371,182 @@ var _ = Describe("Hybrid Translator", func() {
...
-
-					// evaluate results
-					Expect(listener).NotTo(BeNil())
-
-					matchedListeners := listener.GetHybridListener().GetMatchedListeners()
-					Expect(matchedListeners).To(HaveLen(1))
+				Context("HttpConnectionManagerSettings", func() {
+					// while the test for SslConfig seeks to represent a, frankly, hysterical amount of edge conditions, this test
+					// takes a more "throw it at the wall" approach. Notably, it tests more object types than just wrapperspb.BoolValue
+					FIt("Should merge complicated config of various object types with PreventChildOverrides=false", func() {
```
Comment (elcasteel) [inline projects/gateway/pkg/translator/hybrid_translator_test.go:476]: need to unfocus this before merging

**Comment by elcasteel** [comment]
looks good overall

**Review by elcasteel — APPROVED** [review]
